src/cli/cli_history.cpp

```diff
@@ -342,6 +569,56 @@ template <> otError History::Process<Cmd("router")>(Arg aArgs[])
     return error;
 }
 
+/**
+ * @cli history netinfo
+ * @code
+ * history netinfo
+ * | Age                  | Role     | Mode | RLOC16 | Partition ID |
+ * +----------------------+----------+------+--------+--------------+
+ * |         00:00:10.069 | router   | rdn  | 0x6000 |    151029327 |
+ * |         00:02:09.337 | child    | rdn  | 0x2001 |    151029327 |
+ * |         00:02:09.338 | child    | rdn  | 0x2001 |    151029327 |
+ * |         00:07:40.806 | child    | -    | 0x2001 |    151029327 |
+ * |         00:07:42.297 | detached | -    | 0x6000 |            0 |
+ * |         00:07:42.968 | disabled | -    | 0x6000 |            0 |
+ * Done
+ * @endcode
+ * @code
+ * history netinfo list
+ * 00:00:59.467 -> role:router mode:rdn rloc16:0x6000 partition-id:151029327
+ * 00:02:58.735 -> role:child mode:rdn rloc16:0x2001 partition-id:151029327
+ * 00:02:58.736 -> role:child mode:rdn rloc16:0x2001 partition-id:151029327
+ * 00:08:30.204 -> role:child mode:- rloc16:0x2001 partition-id:151029327
+ * 00:08:31.695 -> role:detached mode:- rloc16:0x6000 partition-id:0
+ * 00:08:32.366 -> role:disabled mode:- rloc16:0x6000 partition-id:0
+ * Done
+ * @endcode
+ * @code
+ * history netinfo 2
+ * | Age                  | Role     | Mode | RLOC16 | Partition ID |
+ * +----------------------+----------+------+--------+--------------+
+ * |         00:02:05.451 | router   | rdn  | 0x6000 |    151029327 |
+ * |         00:04:04.719 | child    | rdn  | 0x2001 |    151029327 |
+ * Done
+ * @endcode
+ * @cparam history netinfo [@ca{list}] [@ca{num-entries}]
+ * * Use the `list` option to display the output in list format. Otherwise,
+ *   the output is shown in table format.
+ * * Use the `num-entries` option to limit the output to the number of
+ *   most-recent entries specified. If this option is not used, all stored
+ *   entries are shown in the output.
+ * @par
+ * Displays the network info history in table or list format.
+ * @par
+ * Each table or list entry provides:
+ * * Age: Time elapsed since the command was issued, and given in the format:
+ *        `hours`:`minutes`:`seconds`:`milliseconds`
+ * * Role: Device role. Possible values are `router`, `child`, `detached`, or `disabled`.
+ * * Mode:
```

Comment:
@abtink For "Mode," what are the possible values?

Reply:
Same as previous one.